action/stake_create.go

```diff
@@ -23,6 +25,49 @@ const (
 	CreateStakePayloadGas = uint64(100)
 	// CreateStakeBaseIntrinsicGas represents the base intrinsic gas for CreateStake
 	CreateStakeBaseIntrinsicGas = uint64(10000)
+
+	createStakeInterfaceABI = `[
+		{
+			"inputs": [
+				{
+					"internalType": "string",
+					"name": "candName",
+					"type": "string"
+				},
+				{
+					"internalType": "uint256",
+					"name": "amount",
+					"type": "uint256"
+				},
+				{
+					"internalType": "uint32",
+					"name": "duration",
+					"type": "uint32"
+				},
+				{
+					"internalType": "bool",
+					"name": "autoStake",
+					"type": "bool"
+				},
+				{
+					"internalType": "uint8[]",
+					"name": "data",
+					"type": "uint8[]"
+				}
+			],
+			"name": "createStake",
+			"outputs": [],
+			"stateMutability": "nonpayable",
+			"type": "function"
+		}
+	]`
+)
+
+var (
+	// _createStakeMethod is the interface of the abi encoding of stake action
+	_createStakeMethod abi.Method
+
+	errDecodeFailure = errors.New("failed to decode the data")
```

Comment:
since the logic is exactly same, can add the change for other staking actions in this PR as well